blamo/lib/PDF/pdf_builder.dart

```diff
@@ -10,163 +10,344 @@ import 'package:blamo/Boreholes/BoreholeList.dart';
+            padding: const EdgeInsets.all(5),
+          ),
+          Column(mainAxisAlignment: MainAxisAlignment.center, crossAxisAlignment: CrossAxisAlignment.center,
+            children: widgetTests
+          )
+          ]
+        )
+      )
+    );
+}
+
+List<Level> trimSplitAtIndex(List<Level> splits, int i, double maxHeight){
+  Widget mock;
+  Document canvas = new Document();
+  mock = testsToWidget(splits[i].tests,"");
+  List<Widget> mockWrapper = [mock];
+  //mock.paint(tempContext); 
+  try{
+  canvas.addPage(MultiPage(
+    pageFormat:
+        PdfPageFormat.letter.copyWith(marginBottom: 0.5 * PdfPageFormat.cm,
+                                      marginTop: 0.5 * PdfPageFormat.cm,
+                                      marginLeft: 0.5 * PdfPageFormat.cm,
+                                      marginRight: 0.5 * PdfPageFormat.cm), 
+    crossAxisAlignment: CrossAxisAlignment.start,
+    build: (Context context) => <Widget>[Wrap( 
+              children: mockWrapper)
+    ]));
+  }catch(e){
+
+  }
+  if (mock.box.height > maxHeight){
+    if (i+1 <= splits.length-1){ 
+      // if another free split exists, dump our last test into it.
+       splits[i+1].tests.add(splits[i].tests.last);
+       splits[i].tests.removeLast();
+       splits = trimSplitAtIndex(splits, i, maxHeight);
+    }
+    else{
+      // otherwise we need a new split.
+      splits.add(new Level());
+      splits[i+1].descriptor = splits[i].descriptor;
+      splits[i+1].tests.add(splits[i].tests.last);
+      splits[i].tests.removeLast();
+      splits = trimSplitAtIndex(splits, i, maxHeight);
+    }
+   }
+   else{
+      // touch up the layers visually before we return
+      if(splits[i].descriptor != "Unbounded tests\n"){
+          splits[i].descriptor = "*" + splits[i].descriptor;
+       }
+      List<handler.Test> tempList = [];
+      if(i+1 < splits.length){ // if we have a valid next split to sort, sort it
+        if (splits[i+1].tests.length > 1){
+          for(int x = splits[i+1].tests.length; x > 0; x--){
+            tempList.add(splits[i+1].tests[x-1]);
+          }
+          splits[i+1].tests = tempList;
+        }
+      } 
+      splits[i].scaledRenderHeight = mock.box.height;
+    }
+  return splits;
+}
+
+
+List<Level> boxSplit(Level l, double maxHeight){ 
+  List<Level> splits = [];
+  splits.add(l);
+  for (int i = 0; i < splits.length; i++){
+    if (splits[i].tests.length > 0){
+      splits = trimSplitAtIndex(splits, i, maxHeight);
+    }
+    else{
+      splits.remove(splits[i]);
+    }
+  }
+
+  return splits;
+}
 
 Future<String> docCreate(StateData currentState) async{
 
   // Create levels from provided lists of tests and units 
   var tests = await getTests(currentState);
   var units = await getUnits(currentState);
-  var loginfo = await getLogInfo(currentState.currentDocument, currentState.currentProject);
-
+  var loginfoInit = await getLogInfo(currentState.currentDocument, currentState.currentProject);
```

Comment:
This will need to be changed to just have a value passed in instead of relying on currentState due to the new workflow for export not relying on being in a borehole.Can be updated later though because it is a simple fix.